src/driver/mysql/MysqlDriver.ts

```diff
@@ -217,11 +217,17 @@ export class MysqlDriver implements Driver {
             case ColumnTypes.BOOLEAN:
                 return value === true ? 1 : 0;
             case ColumnTypes.DATE:
-                return moment(value).format("YYYY-MM-DD");
+                if (moment(value).isValid())
+                  return moment(value).format("YYYY-MM-DD");
+                else return '0000-00-00';
```

Comment:
Also I think '0000-00-00' should be never persisted, because if it will be, then after such value will be hydrated new Date('0000-00-00') will create an "invalidate date".